dsb-norge/github-actions
# Maven settings and deploy repos

# Notable changes: maven settings and deploy repos
- new parameter `maven-extra-envs-from-github-yml` makes it possible to export environment variables available to maven with values from the contexts `vars`, `secrets` and `github` in github
- major (non-breaking) changes related to maven settings.xml:
  - no long use `actions/setup-java` to manage maven settings.xml
  - write settings.xml with our own `configure-maven-settings`
    - contents of settings.xml is now configurable through the parameter `maven-user-settings-repositories-yml`
    - it's possible to reference secrets/vars from github in `maven-user-settings-repositories-yml` with the combination of maven's interpolation syntax `${env.MY_ENV}` and new parameter `maven-extra-envs-from-github-yml`. See default values in create-build-envs for example.
- major (non-breaking) changes related to deploy of artifacts with maven:
  - deploy og snapshots/releases is no longer hardcoded to one single maven repo
  - new parameter `maven-build-project-deploy-to-repositories-yml` makes repositories individually configurable for snapshots/releases. Note that credentials for these repos must be configured using the parameter `maven-user-settings-repositories-yml`
- other changes:
  - parameters no longer in use:
    - `maven-repo-pom-id`
    - `maven-repo-username`
    - `maven-repo-token`
    - `maven-build-project-github-packages-enabled`
  - 'create-build-envs':
    - bash test script added to test action locally
  - 'build-maven-project':
    - make sure to unset extra maven environment variables after use so they are not available/exposed to subsequent actions
  - 'build-maven-project', 'create-build-envs' and 'run-maven-command':
    - use `env:` action input for multiline strings in
    - log less when loading helper script
    - bash helper script added

## Other changes
- chore: create-app-vars-matrix: test: support clean and handle errors
- chore: create-app-vars-matrix: log less when loading helper script
- chore: typos and .gitignore
- chore: require-build-envs: use `env:` action input for multiline strings
- chore: deploy-multiple-to-static-is-allowed: use `env:` action input for multiline strings
- chore: deploy-multiple-to-static: use `env:` action input for multiline strings
- chore: build-spring-boot-image: use `env:` action input for multiline strings
- chore: build-nodejs-project: use `env:` action input for multiline strings
